### PR TITLE
Final

### PR DESCRIPTION
**Pull request by AFLLAMA**

### CODE REVIEW

**Comment by AFLLAMA** [comment]
image

**Comment by AFLLAMA** [comment]
In the previous screenshot, the result is for 30 days instead of a decade 
Here is for a decade
image

File: final_project/airflow/dags/process_customers.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        bucket=Variable.get("BUCKET_NAME"),
+        source_objects=[f"{CUSTOMER_FOLDER}{{ ti.xcom_pull(task_ids='determine_latest_date') }}/*.csv"],
+        destination_project_dataset_table="de-07-andrii-feshchenko.bronze.customers",
+        schema_fields=SCHEMA,
+        source_format="CSV",
+        skip_leading_rows=1,
+        gcp_conn_id="my_gcs_connection_id"
+    )
+    query_transform = f"""
+    CREATE OR REPLACE TABLE `{Variable.get("PROJECT_ID")}.silver.customers` AS
```
Comment (tmikhailovska) [inline final_project/airflow/dags/process_customers.py:67]: it's better to put SQL queries in separate files

File: final_project/airflow/dags/process_sales.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            {"name": "Product", "type": "STRING", "mode": "REQUIRED"},
+            {"name": "Price", "type": "STRING", "mode": "REQUIRED"},
+        ],
+        source_format="CSV",
+        skip_leading_rows=1,
+        gcp_conn_id="my_gcs_connection_id"
+    )
+
+    query_transform = f"""
+    CREATE OR REPLACE TABLE `{Variable.get("PROJECT_ID")}.silver.sales` AS
```
Comment (tmikhailovska) [inline final_project/airflow/dags/process_sales.py:40]: the same

File: final_project/airflow/dags/process_user_profiles.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            {"name": "email", "type": "STRING", "mode": "REQUIRED"},
+        ],
+        source_format="NEWLINE_DELIMITED_JSON",
+        skip_leading_rows=1,
+        gcp_conn_id="my_gcs_connection_id"
+    )
+
+    query_transform = f"""
+    CREATE OR REPLACE TABLE `{Variable.get("PROJECT_ID")}.silver.user_profiles` AS
+    SELECT
```
Comment (tmikhailovska) [inline final_project/airflow/dags/process_user_profiles.py:42]: the same

**Review by tmikhailovska — APPROVED** [review]

**Pull request merged**